app/questionnaire/placeholder_transforms.py

```diff
@@ -139,6 +149,79 @@ def calculate_date_difference(first_date: str, second_date: str) -> str:
+        offset_full_weeks: int,
+        days_in_range: int,
+        first_day_of_week: str = "monday",
+    ) -> tuple[str, str]:
+        """Generate a start and end date for a date range given a reference date,
+        weeks prior and number of days in range.
+
+        :param reference_date: The date to reference the date range from, can be YYY&-MM-DD, YY-MM or 'now'.
+        :type reference_date: datetime
+        :param offset_full_weeks: Number of full weeks to offset from the reference date to start the date range.
+        :type offset_full_weeks: int
+        :param days_in_range: Number of days in the range, including the start date, must be a positive integer.
+        :type days_in_range: int
+        :param first_day_of_week: Which day of the week should be considered the first (default 'monday').
+            This is the day of the week which is selected as the start of the range.
+        :type first_day_of_week: str
+        :return: The start and end datetime objects of the range.
+        :rtype: Tuple[datetime, datetime]
+        """
+        reference_datetime = PlaceholderTransforms.parse_date(reference_date)
+        try:
+            first_day_of_week_idx = DAYS[first_day_of_week.lower()]
+        except KeyError as err:
+            raise KeyError(f"'{first_day_of_week}' is not a valid weekday") from err
+
+        if days_in_range < 1:
+            raise ValueError("'days_in_range' must be a positive integer")
+
+        first_day_of_last_partial_week = reference_datetime - timedelta(
+            days=(reference_datetime.weekday() - first_day_of_week_idx) % 7
+        )
+        first_day_of_prior_full_week = first_day_of_last_partial_week + timedelta(
+            days=offset_full_weeks * 7
+        )
+        last_day_of_range = first_day_of_prior_full_week + timedelta(
+            days=days_in_range - 1
+        )
+        return (
+            first_day_of_prior_full_week.strftime(self.input_date_format),
+            last_day_of_range.strftime(self.input_date_format),
+        )
+
+    @staticmethod
+    def format_date_range_pair(date_range: tuple[str, str]) -> str:
+        """Format a pair of dates as a string, clarifying differences in month or year.
+
+        E.g.
+            Monday 1 to Sunday 8 September 2021
+            Monday 29 September to Sunday 6 October 2021
+            Monday 29 December 2021 to Sunday 6 January 2022
+
+        :param date_range: Pair of date strings representing a date range.
+        :type date_range: tuple[str, str]
+        :return: String containing the date range as text.
+        :rtype: str
+        """
+        start_date, end_date = list(map(PlaceholderTransforms.parse_date, date_range))
+        start_date_format = "%A %d"
+        end_date_format = "%A %d %B %Y"
+
+        if start_date.year != end_date.year:
+            start_date_format += " %B %Y"
+        elif start_date.month != end_date.month:
+            start_date_format += " %B"
```

Comment:
On the card checklist, it says `Date format to be <day-of-the-week> <date-of-month> <month> <year> for both from and to`, so is this bit of code relevant anymore?. Worth confirm with Donna what is on the card is still correct or whether they still want this behaviour.

Reply:
Could you update the card accordingly, as that checklist item is misleading?.